alarm-service/src/main/java/com/telecominfraproject/wlan/alarm/controller/AlarmController.java

```diff
@@ -66,6 +68,11 @@ public Alarm create(@RequestBody Alarm alarm ) {
             LOG.error("Failed to create Alarm, request contains unsupported value: {}", alarm);
             throw new DsDataValidationException("Alarm contains unsupported value");
         }
+        
+        if (alarm.getEquipmentId() != 0)
+        {
+            equipmentServiceInterface.get(alarm.getEquipmentId());
```

Comment:
What are we doing with the returned Equipment?

Reply:
nothing, it's just to break the API call if the equipment does not exist

Reply:
i could pull this into a private method with a good name if that helps make it more clear

Reply:
Hmmm. In that case, if the equipment returned is null (or we get a DsEntityNotFoundException), should we throw something as on line 69 above?

Reply:
I don't think it can return null, it should always throw DsEntityNotFoundException.
But yeah i will add a catch for DsEntityNotFoundException and throw DsDataValidationException with a more specific message

Reply:
oh better yet, i will use getOrNull instead and check for null, so i dont have to catch a RuntimeException